lib/fathead/python_coverage/parse.py

```diff
@@ -0,0 +1,282 @@
+        # 4. Empty
+        output_line += "" + "\t"
+        # 5. Categories
+        output_line += "" + "\t"
+        # 6. Empty
+        output_line += "" + "\t"
+        # 7. Related topics
+        output_line += "" + "\t"
+        # 8. Empty
+        output_line += "" + "\t"
+        # 9. External Links (official site) [$url link_text]
+        output_line += "[{} {}]".format(attributes["base_url"],
+                                        attributes["base_title"]) + "\t"
+        # 10. Disambiguation pages only
+        output_line += "" + "\t"
+        # 11. Image
+        output_line += "" + "\t"
+        # 12. Abstract (Main content)
+        output_line += attributes["abstract"] + "\t"
+        # 13. Full URL to source (url with anchor)
+        output_line += attributes["headerlink"] + "\n"
+        return output_line
+
+    def get_parsed_method_from_method_index(self, method_index):
+        """
+        Return parsed method using method index
+        :param method_index: index in method list of method to parse
+        :return: json object of parsed method
+        """
+        raw_method = str(self.get_raw_method(method_index))
+        return self.get_parsed_method_from_data(raw_method)
+
+
+class CoverageParser():
+    def __init__(self,
+                 URL="https://coverage.readthedocs.io/en/latest/api.html"):
+        """
+        Given a URL of the Coverage API, will find
+        and create CoverageClass objects
+        :param URL: URL of entry to CoverageAPI Documentation
+        """
+        self.URL = URL
+        self.classes = []
+        self.api_urls = []
+
+    def get_num_classes(self):
+        return len(self.classes)
+
+    def get_output(self):
+        """
+        Gets output for all methods in class
+        :return: string parsed to output.txt format expected by DDG
+        """
+        output = ""
+        for class_instance in self.classes:
+            method_count = class_instance.get_method_count()
+            for x in range(0, method_count - 1):
+                output += class_instance.get_parsed_method_suitable_for_output(x)
+        return output
+
+    def get_html_data_from_url(self, url):
+        """
+        Parses URL for HTML
+        :param url: URL to be parsed
+        :return: json object of URL and html data
+        """
+        html_data = requests.get(url).text
+        return_data = {
+            "URL": url,
+            "html_data": html_data
+        }
+        return return_data
+
+    def get_html_data_from_file(self, url, filename):
+        """
+        Parses URL for HTML
+        :param url: URL where file retrieved
+        :param filename: filename of html data to parse
+        :return: json object of URL and html data
+        """
+        html_data = open(filename).read()
+        url = url[:url.rindex('/')]
+        if filename.find('/'):
+            filename = filename[filename.rindex('/') + 1:]
+        url = "{}/{}".format(url, filename)
```

Comment:
Hmm, there is 2 issues here with the URLs:

1. The URL is in a list in the output.txt file. ie [https://coverage.readthedocs.io/en/latest/api_coveragedata.html]
2. The list of urls is the following:

```
https://coverage.readthedocs.io/en/latest/api_coverage.html
https://coverage.readthedocs.io/en/latest/api_coverage.htmlThe Coverage class — Coverage.py 4.3.4 documentation
https://coverage.readthedocs.io/en/latest/api_coveragedata.html
https://coverage.readthedocs.io/en/latest/api_coveragedata.htmlThe CoverageData class — Coverage.py 4.3.4 documentation
https://coverage.readthedocs.io/en/latest/api_plugin.html
https://coverage.readthedocs.io/en/latest/api_plugin.htmlPlugin classes — Coverage.py 4.3.4 documentation
https://coverage.readthedocs.io/en/latest/api_plugin.htmlPlugin classes — Coverage.py 4.3.4 documentation
https://coverage.readthedocs.io/en/latest/api_plugin.htmlPlugin classes — Coverage.py 4.3.4 documentation
```

Most of these don't go anywhere and need to link to the original article. So for example, 
[Article] CoveragePlugin sys_info -->  [link] https://coverage.readthedocs.io/en/coverage-4.3.4/api_plugin.html#coverage.CoveragePlugin.sys_info